nomt/src/page_cache.rs

```diff
@@ -146,42 +146,34 @@ impl InflightFetch {
 /// It stores full pages and can be shared between threads.
 #[derive(Clone)]
 pub struct PageCache {
-    shared: Arc<Mutex<Shared>>,
+    shared: Arc<Shared>,
+}
+
+struct Shared {
+    page_rw_pass_domain: RwPassDomain,
     store: Store,
     /// The thread pool used for fetching pages from the store.
     ///
     /// Used for limiting the number of concurrent page fetches.
     fetch_tp: ThreadPool,
-}
-
-struct Shared {
     /// The pages loaded from the store, possibly dirty.
-    cached: HashMap<PageId, Page>,
-    /// The pages that were modified and need to be flushed to the store.
-    ///
-    /// The invariant is that the pages in the dirty set are always present in the `cached` map and
-    /// are marked as dirty there.
-    dirty: HashSet<PageId>,
-    /// Pages that are currently being fetched from the store.
-    inflight: HashMap<PageId, InflightFetch>,
+    cached: DashMap<PageId, PageState>,
```

Comment:
nice, I meant to bring this up today.

though FWIW I think we will need something more specialized. DashMap is just a sharded hash map that partitions the key space into N RwLocked hashmaps.